base/src/test/kotlin/com/r3/corda/ledger/utxo/base/DelegatedContractTests.kt

```diff
@@ -37,28 +37,30 @@ class DelegatedContractTests : ContractTest() {
     fun `verify calls onVerify from the concrete implementation of DelegatedContract`() {
         var onVerifyCalled = false
         val contract = MyDelegatedContract(isVisible = true, { onVerifyCalled = true })
-        val transaction = buildTransaction(NOTARY_KEY, NOTARY_NAME) {
+        val transaction = buildTransaction {
             addCommand(MyVerifiableCommand.One())
-        }
+        }.toLedgerTransaction()
+
         contract.verify(transaction)
```

Comment:
it will also invoke the verify block of the given contract if you replace this two step "verify/assert" with `assertVerifies(transaction)`

Reply:
This one can be replaced with `assertVerifies(transaction)` but most of tests that are left without using `assertVerifies` or `assertFailWith` does not verify as we expected. especially if it's checking something is empty.  e.g. empty states, empty commands. If I understood it correctly, the function [here](https://github.com/corda/corda5-contract-testing/blob/main/contract-testing/src/main/java/com/r3/corda/ledger/utxo/testing/ContractTest.java#L143) wouldn't verify it when things are empty.

Reply:
What that function is doing is invoking the `contract.verify` which is where any checks for empty/null values should occur. Your contract should validate if any states or commands are empty, and error accordingly

Reply:
In case of `verify throws an exception when the permitted command types is empty()`, if assert with `assertFailsWith` it will just succeed the verification where it is exception expected:
```
The transaction was expected to fail, but succeeded.

Expected: "On ''{0}'' contract executing, at least one permitted command type must be specified."
```

Reply:
This suggests your contract verify block isn't behaving as expected

Reply:
But if I understand it correctly, the test should throw exception and it does with the original `assertThrows<IllegalStateException> { contract.verify(transaction) }`

Reply:
For this test class it's easier to leave the verification checks using `contract.verify` and `assertThrows`. This is because we manually create the contract with different parameters to exercise different code paths in `DelegatedContract`. If we use `assertFailsWith`, we'd need to make state classes tied to states with `@BelongsToContract` and then have multiple contracts for each code path.

In this case, I think its simpler to go with Jenny's current implementation.

Reply:
You know your use-case better than me, I was reviewing from the perspective of how the library _should_ be used in normal operation.
If you're happy to go with a custom behaviour, I hope it's clear where the maintenance is.
I just want to check if it's actually easier to create and exercise the contract vs the abstraction, or your just used it.